Makefile

```diff
@@ -1,10 +1,20 @@
+ARCH=
+CC=gcc
 CFLAGS=-O2 -Wall
 DLLINKFLAGS=-Wl,--kill-at -static-libgcc -shared
 
+ifeq ($(shell uname -m),x86_64)
+ifeq ($(ARCH),i686)
+CC = PATH=/mingw32/bin:$$PATH i686-w64-mingw32-gcc -march=i686
+endif
+endif
+
```

Comment:
Since everything worked okay in my 32-bit SDK when hard-coding the cross compiler, I'd rather use that instead.

Besides, this `ARCH=` at the beginning would prevent this `ifeq ($(ARCH), i686)` from working, right?